### ISSUE TITLE
[CLI] Agents hanging up in Sweeps - Linux

### ISSUE THREAD

**Issue by scho97** [wandb/client#2797]
**Description**
Sweep agent hangs up without throwing errors or traceback

**Wandb features**
`wandb.init()`, `wandb.watch()`, `wandb.log()`

**How to reproduce**
I am running two Sweeps on separate GPUs. The first Sweep is working well, but the other one is hanging for a few hours after 66 runs.

I read [#1197](https://github.com/wandb/client/issues/1197) and [#1168](https://github.com/wandb/client/issues/1168) that in Jupyter and Google Colab, you have to move your imports to your train function to solve the problem, and I wanted to ask if it holds also for sweeps that are running in Docker. By moving imports, does it mean I have to put all my `import <LIBRARY>` inside my train function (i.e., `objective_sweep` below)?

Another thing that concerns me is both Sweeps are saving their metadata in `./wandb` directory. Would this be a problem here?

The codes I am using are:

```
import os
...
import torch
import torch.nn as nn
import torch.optim as optim
import wandb

def objective_sweep(args, seed, device):
    # Initialize a new wandb run
    with wandb.init(config=None):
        config = wandb.config # if called by wandb.agent, this config will be set by the Sweep Controller
        ...
        # Train and Validate
        wandb.watch(model, log="all", log_freq=100)

        for epoch in range(start_epoch, args.epochs):

            print('\nEpoch: [%d | %d] LR: %f' % (epoch + 1, args.epochs, config.learning_rate))

            # Train and Validate Network Model
            train_loss, train_acc = train(model, trainloader, criterion, optimizer, epoch, device)
            test_loss, test_acc = test(model, testloader, criterion, device)

            # Log Model Behavior to W&B
            wandb.log({
                "train_loss": train_loss,
                "train_acc": train_acc,
                "test_loss": test_loss,
                "test_acc": test_acc,
                "best_acc": best_acc,
                "epoch": epoch
            })
       ...
       print('Best accuracy: {}'.format(best_acc))

    return None
```
```
# Configure Settings
study_name = os.path.basename(os.path.normpath(args.checkpoint))
entity_name = args.entity_name
sweep_config = create_sweep_config(args,study_name,verbose=True)
# Initialize Sweep: Wind Up the Sweep Controller
sweep_id = wandb.sweep(sweep=sweep_config, project=study_name, entity=entity_name)
# Run Sweep Agent
wandb.agent(sweep_id, function=lambda: objective_sweep(args, seed=SEED, device=DEVICE), entity=entity_name, project=study_name, count=N_TRIAL)
```
**Environment**
- OS: Mac OS Big Sur 11.6
- Environment: Linux, Docker
- Python Version: 3.6.9

**Comment by vanpelt** [comment]
If you're running in Docker, you should likely use our command line agent.  The import trick is only for jupyter environments that will execute your code in the same python context.  You can run:

```
wandb agent ENTITY/PROJECT/SWEEP_ID --count=N_TRIAL
```

And then the agent will run whatever "program" you specified in your sweep config.  That's much simpler in the docker context and will mean you don't need to worry about shuffling your imports.

**Comment by vanpelt** [comment]
Hangs should generally only happen if a single python process launches multiple training jobs.  By running it from the command line each training job is guarenteed to be a new process.
